typeform/typeform.py

```diff
@@ -0,0 +1,183 @@
+
+
+        forms = source.get('forms')
+        forms = forms if forms else []
+
+        # add an 'offset' attribute used for pagination for each
+        # different form, used as the actual number to use as offset
+        self._forms = map(lambda form: {
+            'id': form['id'],
+            'name': form['name'],
+            'offset': 0
+        }, forms)
+
+        self._key = source.get('key')
+        self._total = len(self._forms)
+
+    def read(self, n = None):
+        if len(self._forms) == 0:
+            # no more data to consume
+            return None
+
+        form = self._forms[0]
+        url = '%s/form/%s?key=%s&completed=true&offset=%s&limit=%s' % (
+            BASE_URL,
+            form['id'],
+            self._key,
+            form['offset'],
+            FETCH_LIMIT
+        )
+
+        # pull data incrementally if configured to do so.
+        if self._incval:
+            url += '&since=%s' % (self._incval) 
+
+        body = self._request(url)
+
+        # 'completed' represent the number of completed forms that
+        # returned from the server
+        stats = body.get('stats').get('responses')
+        total = stats['completed']
+
+        if (form['offset'] + FETCH_LIMIT) >= total:
+            # we're done paginating with the current form.
+            form['offset'] = None
+
+            # no more results for this form, remove it
+            self._forms.pop(0)
+
+            # report progress
+            loaded = self._total - len(self._forms)
+            msg = '%s of %s forms fetched' % (loaded, self._total)
+            self.progress(loaded, self._total, msg)
+        else:
+            # prepare the offset to the next set of records
+            form['offset'] += FETCH_LIMIT
+
+
+        def add_dest(dest):
+            def add(item):
+                item['__table'] = dest
+                return item
+
+            return add
+
+        # clean up irrelevant attributes and add the
+        # destination table name
+        # 
+        # add the questions records
+        dest = form['name'] + '_questions'
+        results = map(add_dest(dest), body['questions'])
+
+        # add the responses (answers) records
+        dest = form['name'] + '_responses'
+        responses = map(add_dest(dest), body['responses'])
+
+        results.extend(responses)
+        return results
+
+
+    # GET all the forms.
+    def get_forms(self):
+        # provides us with all of the user forms
+        url = '%s/forms?key=%s' % (
+            BASE_URL,
+            self._key
+        )
+
+        # the body of the result is a list of forms
+        return self._request(url)
+
+    # Helper function for issuing GET requests 
+    def _request(self, url):
+        self.log('GET', url)
+
+        try:
+            res = urllib2.urlopen(url)
+        except urllib2.HTTPError, e:
+            raise TypeformError.from_http_error(e)
+
+        self.log('RESPONSE', url)
```

Comment:
Not sure of the purpose of this line... logging the `url` again?
Think its unnecessary